src/v1.ts

```diff
@@ -17,25 +17,43 @@ function v1(token: string) {
     per_page: 50,
     sort: 'asc',
     fields: ['name'],
-  }
+  };
 
   const getAllBoards = async (options?: GetAllBoardOptions): Promise<[Board]> => {
     const mergedOptions = {
       ...defaultGetAllBoardOptions,
       ...options
-    }
+    };
     return (
       await axios.get(`/glo/boards?page=${mergedOptions.page}&per_page=${mergedOptions.per_page}&archived=${mergedOptions.archived}&sort=${mergedOptions.sort}&fields=${mergedOptions.fields.join('%2C')}`)
     ).data;
-  }
+  };
 
   const boards = {
+    getAll: getAllBoards,
     get: async (board_id: string, options?: GetBoardOptions): Promise<Board> => {
       return (
         await axios.get(`/glo/boards/${board_id}?fields=${ ((options && options.fields) || ['name']).join('%2C')}`
         )
       ).data;
     },
+    create: async (board_name: string): Promise<Board> => {
+      return (await axios.post(`/glo/boards`, { name: board_name })).data;
+    },
+    delete: async (board_id: string): Promise<any> => {
+      return (await axios.delete(`/glo/boards/${board_id}`)).data;
+    },
+    labels: {
+      create: async (board_id: string, label: Label): Promise<Label> => {
```

Comment:
My concern is that, according to the [API docs](https://gloapi.gitkraken.com/v1/docs/#/Labels/post_boards__board_id__labels), the name and color fields are always required for creating labels. However, when querying a board, all the fields on label can be toggled, which is why James made them all optional in the Label type. We should declare a different type (maybe call it NewLabel) that enforces this behavior.
```typescript
// src/v1_types.ts
export type NewLabel = Label & {
    color: Color;
    name: string;
}
```
```suggestion
      create: async (board_id: string, label: NewLabel): Promise<Label> => {
```